core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/MetadataServiceTest.java

```diff
@@ -136,9 +139,9 @@ public void testGetMetadataDatasourceIds() throws Exception {
     mockSet.add( "domainId1" );
     mockMetadataIdsList.add( "domainId1" );
 
-    doReturn( true ).when( metadataService ).isMetadataDatasource( "domainId1" );
+    doReturn( 1 ).when( metadataService ).noOfThreads();
+    doReturn( "domainId1" ).when( metadataService ).isMetaDataSource( "domainId1" );
```

Comment:
similar comment to as `[DataSourceWizardServiceTest#testGetDSWDatasourceIds()`.
I don't think this is a valid unit test, when you mock out functions for the class under test.

I would suggest:

1. don't mock `MetadataService#isMetaDataSource(string)`, instead mock the call to `metadataDomainRepository.getDomain( id );` and create an Object Domain that will pass.